packages/shared_preferences/shared_preferences/lib/shared_preference_async.dart

```diff
@@ -0,0 +1,313 @@
+  ///
+  /// If no [parameters] are provided, and [SharedPreferencesAsync] has no filter,
+  /// all preferences will be removed. This includes anything not set by this plugin,
+  /// which may create some unwanted behaviors. It is highly recommended that
+  /// [PreferencesFilters] be provided to this call.
+  Future<bool> clear(ClearPreferencesParameters parameters) {
+    return _platform.clear(parameters, _options);
+  }
+}
+
+/// Options necessary to create a [SharedPreferencesWithCache].
+class SharedPreferencesWithCacheOptions {
+  /// Creates a new instance with the given options.
+  const SharedPreferencesWithCacheOptions({
+    required this.filter,
+  });
+
+  /// Information about what data should be fetched during `getAll` and `init`
+  /// methods, as well as what data will be removed by `clear`.
+  final PreferencesFilters filter;
+}
+
+/// Provides a persistent store for simple data.
+///
+/// Cache provided to allow for synchronous gets.
+///
+/// If preferences on the platform may be altered by other means than through
+/// this plugin, consider using [SharedPreferencesAsync] instead. You may also
+/// refresh the cached data using [reloadCache] prior to a get request to prevent
+/// missed changes that may have occurred since the cache was last updated.
+class SharedPreferencesWithCache {
+  /// Creates a new instance with the given options.
+  SharedPreferencesWithCache._create({
+    required this.sharedPreferencesOptions,
+    required SharedPreferencesWithCacheOptions cacheOptions,
+    Map<String, Object?>? cache,
+  })  : _cacheOptions = cacheOptions,
+        _directAccess =
+            SharedPreferencesAsync(options: sharedPreferencesOptions),
+        _cache = cache ?? <String, Object?>{};
+
+  /// Creates a new instance with the given options and reloads the cache from
+  /// the platform data.
+  static Future<SharedPreferencesWithCache> create({
+    required SharedPreferencesOptions sharedPreferencesOptions,
+    required SharedPreferencesWithCacheOptions cacheOptions,
+    Map<String, Object?>? cache,
+  }) async {
+    final SharedPreferencesWithCache preferences =
+        SharedPreferencesWithCache._create(
+      sharedPreferencesOptions: sharedPreferencesOptions,
+      cacheOptions: cacheOptions,
+      cache: cache,
+    );
+
+    await preferences._init();
+
+    return preferences;
+  }
+
+  /// Cache containing in-memory data.
+  final Map<String, Object?> _cache;
+
+  /// Options that determine the behavior of contained methods, usually
+  /// platform specific extensions of the [SharedPreferencesOptions] class.
+  final SharedPreferencesOptions sharedPreferencesOptions;
+
+  /// Options that define cache behavior.
+  final SharedPreferencesWithCacheOptions _cacheOptions;
+
+  /// Async access directly to the platform.
+  ///
+  /// Methods called through [_directAccess] will NOT update the cache.
+  final SharedPreferencesAsync _directAccess;
+
+  Future<void> _init() async {
```

Comment:
Why the wrapper, vs just calling `reloadCache()` in `create`?

Reply:
Style points.